src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -190,7 +185,16 @@ private void ClearStore()
             taintStore.Set(Enumerable.Empty<IAnalysisIssueVisualization>(), null);
         }
 
-        private void UpdateTaintIssuesUIContext(bool hasTaintIssues) =>
-            vsMonitorSelection.SetCmdUIContext(contextCookie, hasTaintIssues ? 1 : 0);
+        private void UpdateTaintIssuesUIContext(bool hasTaintIssues)
```

Comment:
Was able to move everything in here, I don't think calling getcmdcookie multiple times should be a problem

Reply:
Probably not... does VS return the same value for the cookie on each call to `GetCmdUIContextCookie`? If so, then it should be ok.

Reply:
Here it says "Note Once a GUID is registered through this method, the GUID remains registered until the IDE exits." So from my understanding yes it always returns the same one

https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivsmonitorselection.getcmduicontextcookie?view=visualstudiosdk-2022